### PR TITLE
Add Tool.programmingLanguages

### LINKED ISSUE #256: Capture information about the programming language(s) used by a tool
All the tools are dockerized and can be run without knowing the programming languages used internally. However, the programming language(s) used by a tool by help developers to identify whether they want to contribute to an existing tool or extend one.

- We could support the list of [programming languages identified by PyPi](https://pypi.org/search/?c=Programming+Language+%3A%3A+APL).
- Here is [a list of languages "supported" by GitHub](https://docs.github.com/en/get-started/learning-about-github/github-language-support)

### PR DESCRIPTION
**Pull request by tschaffter**
Fixes #256 

## Notes

- The proposed list started from the [one identified by PyPi](https://pypi.org/search/?c=Programming+Language+%3A%3A+APL) + popular languages I recognized from the [TIOBE index](https://www.tiobe.com/tiobe-index/).
- Specifying the new property `Tool.programmingLanguages` is optional to lower the requirements required for submitting a valid tool.

### CODE REVIEW

**Review by thomasyu888 — APPROVED** [review]

File: openapi/commons/components/schemas/Tool.yaml [hunk 1]
```diff
@@ -39,6 +39,13 @@ properties:
     type: string
     minLength: 1
     pattern: "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+  programmingLanguages:
+    description: The programming languages used to develop this tool
+    type: array
+    items:
+      $ref: ../parameters/ProgrammingLanguage.yaml
+    minItems: 1
```
Comment (cascadianblue) [inline openapi/commons/components/schemas/Tool.yaml:47]: I am -1 on making this required (i.e. I think `minItems` should be `0`)

Comment (tschaffter) [reply openapi/commons/components/schemas/Tool.yaml:47]: @cascadianblue Can you explain why you think it's best to not require the developer to specify at least one language?

Comment (tschaffter) [reply openapi/commons/components/schemas/Tool.yaml:47]: @cascadianblue I agree now with your suggestion. At this early stage of the project, we want to make it easier as possible for participants to submit their tool. If the NLP Sandbox get widely adopted, then we can start enforcing participants to provide extra metadata.

**Comment by tschaffter** [comment]
@gkowalski @ymnliu Could you please share your thoughts on this proposal?

**Comment by gkowalski** [comment]
I as a developer feel that having this baked into the tool is overkill.  I'd rather use the "url" already included to go back to the submitting projects Homepage where it it discussed what NLP Libraries are used. Mary libraries / trained models can be used across multiple languages. Of course I'm not looking at it from a Synapse point of view where fostering collaboration  and metrics of what languages used may be a goal. If needed , then I suggest having it "optional" to the API.

**Review by gkowalski — APPROVED** [review]
As long as it's optional I'm ok with it.

**Pull request merged**